lib/posix/options/pipe.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) Bruce ROSIER
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "posix_features.h"
+#include "zephyr/fs/fs.h"
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/sys/fdtable.h>
+#include <sys/_intsup.h>
+
+#define NUM_PIPES        CONFIG_POSIX_NUM_PIPES
+#define PIPE_BUFFER_SIZE _POSIX_PIPE_BUF
```

Comment:
this seems to be overuse of macros. Maybe consider just using `CONFIG_POSIX_PIPE_BUF`?